### PR TITLE
[Torch 2.0 compile] Fix more torch compile breaks

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Follow-up PR from: https://github.com/huggingface/diffusers/pull/3286

It would be great if we could benchmark the following scripts
for both torch 2.0.0 and torch nightly.

**NOTE: ALL THE FOLLOWING SCRIPTS CAN EASILY BE ACCESSED HERE: https://huggingface.co/diffusers/tools/tree/main/benchmark**

## 1. Stable Diffusion - Text2Img

```py
from diffusers import DiffusionPipeline
import torch

path = "runwayml/stable-diffusion-v1-5"

run_compile = True  # Set True / False

pipe = DiffusionPipeline.from_pretrained(path, torch_dtype=torch.float16)
pipe = pipe.to("cuda:0")
pipe.unet.to(memory_format=torch.channels_last)

if run_compile:
    print("Run torch compile")
    pipe.unet = torch.compile(pipe.unet, mode="reduce-overhead", fullgraph=True)

prompt = "ghibli style, a fantasy landscape with castles"

for _ in range(3):
    images = pipe(prompt=prompt).images
```

## 2.  Stable Diffusion - Img2Img

```py
from diffusers import StableDiffusionImg2ImgPipeline
import requests
import torch
from PIL import Image
from io import BytesIO

url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"

response = requests.get(url)
init_image = Image.open(BytesIO(response.content)).convert("RGB")
init_image = init_image.resize((512, 512))

path = "runwayml/stable-diffusion-v1-5"

run_compile = True  # Set True / False

pipe = StableDiffusionImg2ImgPipeline.from_pretrained(path, torch_dtype=torch.float16)
pipe = pipe.to("cuda:0")
pipe.unet.to(memory_format=torch.channels_last)

if run_compile:
    print("Run torch compile")
    pipe.unet = torch.compile(pipe.unet, mode="reduce-overhead", fullgraph=True)

prompt = "ghibli style, a fantasy landscape with castles"

for _ in range(3):
    image = pipe(prompt=prompt, image=init_image).images[0]
```

## 3.  Stable Diffusion - InPaint

```py
from diffusers import StableDiffusionInpaintPipeline
import requests
import torch
from PIL import Image
from io import BytesIO

url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"

def download_image(url):
    response = requests.get(url)
    return Image.open(BytesIO(response.content)).convert("RGB")


img_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo.png"
mask_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo_mask.png"

init_image = download_image(img_url).resize((512, 512))
mask_image = download_image(mask_url).resize((512, 512))

path = "runwayml/stable-diffusion-inpainting"

run_compile = True  # Set True / False

pipe = StableDiffusionInpaintPipeline.from_pretrained(path, torch_dtype=torch.float16)
pipe = pipe.to("cuda:0")
pipe.unet.to(memory_format=torch.channels_last)

if run_compile:
    print("Run torch compile")
    pipe.unet = torch.compile(pipe.unet, mode="reduce-overhead", fullgraph=True)

prompt = "ghibli style, a fantasy landscape with castles"

for _ in range(3):
    image = pipe(prompt=prompt, image=init_image, mask_image=mask_image).images[0]
```

## 4. ControlNet

```py
from diffusers import StableDiffusionControlNetPipeline, ControlNetModel
import requests
import torch
from PIL import Image
from io import BytesIO

url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"

response = requests.get(url)
init_image = Image.open(BytesIO(response.content)).convert("RGB")
init_image = init_image.resize((512, 512))

path = "runwayml/stable-diffusion-v1-5"

run_compile = True  # Set True / False
controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny", torch_dtype=torch.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained(
    path, controlnet=controlnet, torch_dtype=torch.float16
)

pipe = pipe.to("cuda:0")
pipe.unet.to(memory_format=torch.channels_last)
pipe.controlnet.to(memory_format=torch.channels_last)

if run_compile:
    print("Run torch compile")
    pipe.unet = torch.compile(pipe.unet, mode="reduce-overhead", fullgraph=True)
    pipe.controlnet = torch.compile(pipe.controlnet, mode="reduce-overhead", fullgraph=True)

prompt = "ghibli style, a fantasy landscape with castles"

for _ in range(3):
    image = pipe(prompt=prompt, image=init_image).images[0]
```

## 5. DeepFloyd IF

```py
from diffusers import DiffusionPipeline
import torch

run_compile = True  # Set True / False

pipe = DiffusionPipeline.from_pretrained("DeepFloyd/IF-I-M-v1.0", variant="fp16", text_encoder=None, torch_dtype=torch.float16)
pipe.to("cuda")
pipe_2 = DiffusionPipeline.from_pretrained("DeepFloyd/IF-II-M-v1.0", variant="fp16", text_encoder=None, torch_dtype=torch.float16)
pipe_2.to("cuda")
pipe_3 = DiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-x4-upscaler", torch_dtype=torch.float16)
pipe_3.to("cuda")


pipe.unet.to(memory_format=torch.channels_last)
pipe_2.unet.to(memory_format=torch.channels_last)
pipe_3.unet.to(memory_format=torch.channels_last)

if run_compile:
    pipe.unet = torch.compile(pipe.unet, mode="reduce-overhead", fullgraph=True)
    pipe_2.unet = torch.compile(pipe_2.unet, mode="reduce-overhead", fullgraph=True)
    pipe_3.unet = torch.compile(pipe_3.unet, mode="reduce-overhead", fullgraph=True)

prompt = "the blue hulk"

prompt_embeds = torch.randn((1, 2, 4096), dtype=torch.float16)
neg_prompt_embeds = torch.randn((1, 2, 4096), dtype=torch.float16)

for _ in range(3):
    image = pipe(prompt_embeds=prompt_embeds, negative_prompt_embeds=neg_prompt_embeds, output_type="pt").images
    image_2 = pipe_2(image=image, prompt_embeds=prompt_embeds, negative_prompt_embeds=neg_prompt_embeds, output_type="pt").images
    image_3 = pipe_3(prompt=prompt, image=image, noise_level=100).images
```

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
# 1. First benchmark on GPU RTX4090 (in it/s):

| Model      | torch 2.0 - no compile | torch nightly - no compile | torch 2.0 - compile | torch nightly - compile |
|------------|------------------------|-----------------------------|----------------------|---------------------------|
| SD - txt2img |         :x:            |             41.37it/s             |         :x:          |             48.96it/s           |
| SD - img2img |         :x:            |             41.18it/s             |         :x:          |             49.03it/s            |
| SD - inpaint |         :x:            |             41.49it/s             |         :x:          |             49.16it/s            |
| SD - controlnet|       :x:            |             29.80it/s             |         :x:          |             35.50it/s            |
| IF         |         :x:            |             69.03it/s / 18.80it/s / 84.91it/s             |         :x:          |             125.45it/s / 26.47it/s / 138.82it/s            |

**Comment by patrickvonplaten** [comment]
@sayakpaul just a heads-up - this branch will soon be used for the main IF space: https://huggingface.co/spaces/DeepFloyd/IF/discussions/136

**Comment by sayakpaul** [comment]
@patrickvonplaten I slightly edited https://github.com/huggingface/diffusers/pull/3313#issuecomment-1531236397 to correct the installation of PyTorch 2.0. Previously, the wheel was actually pointing to nightly for the stable release.

**Comment by sayakpaul** [comment]
More benchmarks!

## Setting up the environment

First, I would like to shed light on how I set up the environment, as it appeared to be more involed than I'd imagined. 

1. I started with `nvidia-docker run -it --rm nvidia/cuda:11.8.0-runtime-ubuntu18.04`. 
2. After I was in the Docker image, I set up the environment as follows:

  ```bash
   apt-get update && apt-get install -y git \
    make build-essential libssl-dev zlib1g-dev \
    libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev git-lfs  \
    ffmpeg libsm6 libxext6 cmake libgl1-mesa-glx \
    && rm -rf /var/lib/apt/lists/* \
    && git lfs install

 curl https://pyenv.run | bash

 export PATH=$HOME/.pyenv/shims:$HOME/.pyenv/bin:$PATH

 pyenv install 3.10 && pyenv global 3.10 && pyenv rehash

 pip install --no-cache-dir --upgrade pip==22.3.1 setuptools wheel
 
 # Change this to the stable 11.8 release as necessary
 # as shown in the comments above.
 pip install torch --index-url https://download.pytorch.org/whl/nightly/cu118

 pip install git+https://github.com/huggingface/diffusers@fix_torch_compile accelerate transformers sentencepiece
```

3. Then I was able to run the benchmarks from https://huggingface.co/diffusers/tools/tree/main/benchmark as shared by @patrickvonplaten in https://github.com/huggingface/diffusers/pull/3313#issue-1691984499. 

## Notes

* I spun up three different machines (as in with different GPUs, T4, V100, A100, in this case). 
* When I ran into the following issue, I uninstall `pytorch-triton` and `triton` and installed `triton` again with `pip install triton`. 

  ```bash
  torch._dynamo.exc.BackendCompilerFailed: debug_wrapper raised RuntimeError: Cannot 
  find a working triton installation. More information on installing Triton can be 
  found at https://github.com/openai/triton
  ```

Now coming to the numbers:

## A100

| **Model** | **torch 2.0 - <br>no compile** | **torch nightly - <br>no compile** | **torch 2.0 - <br>compile** | **torch nightly - <br>compile** |
|:---:|:---:|:---:|:---:|:---:|
| SD - txt2img | 21.66it/s | 23.13it/s | 44.03it/s | 49.74it/s |
| SD - img2img | 21.81it/s | 22.40it/s | 43.92it/s | 46.32it/s |
| SD - inpaint | 22.24it/s | 23.23it/s | 43.76it/s | 49.25it/s |
| SD - controlnet | 15.02it/s | 15.82it/s | 32.13it/s | 36.08it/s |
| IF | 20.21it/s / <br>13.84it/s / <br>24.00it/s | 20.12it/s / <br>13.70it/s / <br>24.03it/s | ❌ | 97.34it/s / <br>27.23it/s / <br>111.66it/s |

## V100

| **Model** | **torch 2.0 - <br>no compile** | **torch nightly - <br>no compile** | **torch 2.0 - <br>compile** | **torch nightly - <br>compile** |
|:---:|:---:|:---:|:---:|:---:|
| SD - txt2img | 18.99it/s | 19.14it/s | 20.95it/s | 22.17it/s |
| SD - img2img | 18.56it/s | 19.18it/s | 20.95it/s | 22.11it/s |
| SD - inpaint | 19.14it/s | 19.06it/s | 21.08it/s | 22.20it/s |
| SD - controlnet | 13.48it/s | 13.93it/s | 15.18it/s | 15.88it/s |
| IF |  20.01it/s / <br>9.08it/s / <br>23.34it/s | 19.79it/s / <br>8.98it/s / <br>24.10it/s | ❌ | 55.75it/s / <br>11.57it/s / <br>57.67it/s |

## T4

Incoming

**Comment by patrickvonplaten** [comment]
Cool let's merge this and update the docs in a follow-up PR :-) (cc @sayakpaul)

File: src/diffusers/models/controlnet.py [hunk 1]
```diff
@@ -517,7 +517,7 @@ def forward(
 
         controlnet_cond = self.controlnet_cond_embedding(controlnet_cond)
 
-        sample += controlnet_cond
+        sample = sample + controlnet_cond
```
Comment (pcuenca) [inline src/diffusers/models/controlnet.py:520]: How does this break compilation? I don't get it 🤯

Comment (Chillee) [reply src/diffusers/models/controlnet.py:520]: I don't think this is necessary for `torch.compile` on nightly. I'm guessing on an older release it was just breaking for no good reason :)

File: tests/pipelines/stable_diffusion/test_stable_diffusion_controlnet.py [hunk 2]
```diff
@@ -585,6 +586,42 @@ def test_canny_guess_mode(self):
...
+        if version.parse(torch.__version__) < version.parse("2.0"):
+            print(f"Test `test_stable_diffusion_ddim` is skipped because {torch.__version__} is < 2.0")
+            return
+
+        controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny")
+
+        pipe = StableDiffusionControlNetPipeline.from_pretrained(
+            "runwayml/stable-diffusion-v1-5", safety_checker=None, controlnet=controlnet
+        )
+        pipe.to("cuda")
```
Comment (shingjan) [inline tests/pipelines/stable_diffusion/test_stable_diffusion_controlnet.py:599]: does this pass test? I am getting a 
```
  File "/home/yj/pytorch/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/yj/diffusers/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py", line 892, in __call__
    self.check_inputs(
  File "/home/yj/diffusers/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py", line 984, in <resume in __call__>
    latents = self.prepare_latents(
  File "/home/yj/diffusers/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py", line 734, in prepare_latents
    latents = randn_tensor(shape, generator=generator, device=device, dtype=dtype)
  File "/home/yj/diffusers/src/diffusers/utils/torch_utils.py", line 75, in randn_tensor
    latents = torch.randn(shape, generator=generator, device=rand_device, dtype=dtype, layout=layout).to(device)
RuntimeError: Expected a 'cuda' device type for generator but found 'cpu'
```
after rebasing with this change. Thanks!

**Comment by dblunk88** [comment]
Getting this error:
  0%|          | 0/100 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/david/PycharmProjects/local_diffusion/deepfloyd.py", line 61, in <module>
    image = pipe(
  File "/home/david/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/david/PycharmProjects/local_diffusion/diffusers/src/diffusers/pipelines/deepfloyd_if/pipeline_if.py", line 791, in __call__
    noise_pred = self.unet(
  File "/home/david/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 280, in _fn
    return fn(*args, **kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 433, in catch_errors
    return callback(frame, cache_size, hooks, frame_state)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 122, in _fn
    return fn(*args, **kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 337, in _convert_frame_assert
    return _compile(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 177, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 407, in _compile
    out_code = transform_code_object(code, transform)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 1000, in transform_code_object
    transformations(instructions, code_options)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 392, in transform
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2006, in run
    super().run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1135, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars.items)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 306, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 269, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 102, in call_function
    return tx.inline_user_function_return(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 590, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2111, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2189, in inline_call_
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1095, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 269, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 102, in call_function
    return tx.inline_user_function_return(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 590, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2111, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2189, in inline_call_
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1147, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 269, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 102, in call_function
    return tx.inline_user_function_return(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 590, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2111, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2189, in inline_call_
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1095, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 102, in call_function
    return tx.inline_user_function_return(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 590, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2111, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2189, in inline_call_
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1135, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars.items)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 269, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 102, in call_function
    return tx.inline_user_function_return(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 590, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2111, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2189, in inline_call_
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1095, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 102, in call_function
    return tx.inline_user_function_return(
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 590, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2111, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 2189, in inline_call_
    tracer.run()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 703, in run
    and self.step()
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 663, in step
    getattr(self, inst.opname)(inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 385, in wrapper
    return inner_fn(self, inst)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1095, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 554, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 584, in call_function
    result = handler(tx, *args, **kwargs)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/builtin.py", line 941, in call_hasattr
    return obj.call_hasattr(tx, name).add_options(self, obj, attr)
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 225, in call_hasattr
    unimplemented(f"hasattr: {repr(self)}")
  File "/home/david/.local/lib/python3.10/site-packages/torch/_dynamo/exc.py", line 133, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: hasattr: TensorVariable()

from user code:
   File "/home/david/.local/lib/python3.10/site-packages/accelerate/utils/operations.py", line 100, in recursively_apply
    elif test_type(data):
  File "/home/david/.local/lib/python3.10/site-packages/accelerate/utils/operations.py", line 131, in _has_to_method
    return hasattr(t, "to")

Set torch._dynamo.config.verbose=True or TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True

**Comment by sayakpaul** [comment]
@dblunk88 could you maybe open a new issue for this? Easier for us to track.

File: src/diffusers/models/controlnet.py (hunk 1 above)
Comment (pcuenca) [reply src/diffusers/models/controlnet.py:520]: Thanks for the clarification!

**Pull request merged**